docs/DesignDocs/RemoteAttestationCollaterals.md

```diff
@@ -0,0 +1,457 @@
+ * @param[in] opt_params Optional additional parameters needed for the current
+ * enclave type.
+ *     For SGX:
+ *        This can be sgx_target_info_t for local attestation.
+ * @param[in] opt_params_size The size of the **opt_params** buffer.
+ * @param[out] evidence_buffer This points to the resulting evidence upon success.
+ * @param[out] evidence_buffer_size This is set to the size of the evidence buffer
+ * on success.
+ * @param[out] endorsements_buffer The buffer containing the endorsements to parse.
+ * @param[out] endorsements_buffer_size The size of the **endorsements_buffer**.
+ *
+ * @retval OE_OK The evidence and endorsements were successfully created.
+ * @retval OE_INVALID_PARAMETER At least one parameter is invalid.
+ * @retval OE_OUT_OF_MEMORY Failed to allocate memory.
+ *
+ */
+oe_result_t oe_get_evidence(
+    uint32_t flags,
+    const uint8_t* custom_claims,
+    size_t custom_claims_size,
+    const void* opt_params,
+    size_t opt_params_size,
+    uint8_t** evidence_buffer,
+    size_t* evidence_buffer_size,
+    uint8_t** endorsements_buffer,
+    size_t* endorsements_buffer_size);
+
+/**
+ * Free up any resources allocated by oe_get_evidence()
+ *
+ * @param[in] evidence_buffer THe buffer containing the evidence.
+ * @param[in] endorsements_buffer The buffer containing the endorsements.
+ */
+void oe_free_evidence(
+    uint8_t* endorsements_buffer,
+    uint8_t* evidence_buffer);
+
+/**
+ * Verify the integrity of the evidence and its signature,
+ * with endorsements that are associated with the evidence.
+ * This function works for both local and remote attestation.
+ *
+ * This function is available in the enclave as well as in the host.
+ *
+ * @param[in] evidence_buffer The buffer containing the evidence to verify.
+ * @param[in] evidence_buffer_size The size of the **evidence** buffer.
+ * @param[in] endorsements Optional The endorsement data that is associated with
+ * the evidence.
+ * @param[in] endorsements_size The size of the **endorsements** buffer.
+ * @param[in] input_validation_time Optional datetime to use when verifying
+ * evidence. If not specified, it will use the creation_datetime of the
+ * endorsements (if any endorsements are provided).
+ * @param[out] claims The list of claims.
+ * @param[out] claims_size The size of claims.
+ *
+ * @retval OE_OK The verification was successful.
+ * @retval OE_INVALID_PARAMETER At least one parameter is invalid.
+ *
+ */
+oe_result_t oe_verify_evidence(
+    const uint8_t* evidence_buffer,
+    size_t evidence_buffer_size,
+    const uint8_t* endorsements,
+    size_t endorsements_size,
+    oe_datetime_t* input_validation_time,
+    uint8_t** claims,
+    size_t* claims_size);
+```
+
+### Claims
+As part of the claims form `oe_verify_evidence()`, there will be a validity datetime range that applies to the evidence and endorsements.  For more information see the design document `CustomAttestation.md`.
```

Comment:
Claims are a key part of this API design, but is not included in this PR? And we're checking in an implementation of this at the same time as the design approval?

Please refactor this PR apart so the design can be evaluated as a whole, not as a partial design + implementation.

Reply:
My plan is to create a separate PR.